JitsiConferenceEventManager.js

```diff
@@ -0,0 +1,495 @@
+        function (devices) {
+            conference.room.updateDeviceAvailability(devices);
+        });
+};
+
+/**
+ * Setups event listeners related to conference.xmpp
+ */
+JitsiConferenceEventManager.prototype.setupXMPPListeners = function () {
+    var conference = this.conference;
+    conference.xmpp.addListener(
+        XMPPEvents.CALL_INCOMING, conference.onIncomingCall.bind(conference));
+
+    conference.xmpp.addListener(XMPPEvents.START_MUTED_FROM_FOCUS,
+        function (audioMuted, videoMuted) {
+            conference.startAudioMuted = audioMuted;
+            conference.startVideoMuted = videoMuted;
+
+            // mute existing local tracks because this is initial mute from
+            // Jicofo
+            conference.getLocalTracks().forEach(function (track) {
+                if (conference.startAudioMuted && track.isAudioTrack()) {
+                    track.mute();
+                }
+                if (conference.startVideoMuted && track.isVideoTrack()) {
+                    track.mute();
+                }
+            });
+
+            conference.eventEmitter.emit(JitsiConferenceEvents.STARTED_MUTED);
+        });
+};
+
+/**
+ * Setups event listeners related to conference.statistics
+ */
+JitsiConferenceEventManager.prototype.setupStatisticsListeners = function () {
+    var conference = this.conference;
+    if(!conference.statistics)
+        return;
+
+    conference.statistics.addAudioLevelListener(function (ssrc, level) {
+        var userId = null;
+
+        var resource = conference.rtc.getResourceBySSRC(ssrc);
+        if (!resource)
+            return;
+
+        conference.rtc.setAudioLevel(resource, level);
+    });
+    conference.statistics.addConnectionStatsListener(function (stats) {
+        var ssrc2resolution = stats.resolution;
+
+        var id2resolution = {};
+
+        // preprocess resolutions: group by user id, skip incorrect
+        // resolutions etc.
+        Object.keys(ssrc2resolution).forEach(function (ssrc) {
+            var resolution = ssrc2resolution[ssrc];
+
+            if (!resolution.width || !resolution.height ||
+                resolution.width == -1 || resolution.height == -1) {
+                return;
+            }
+
+            var id = conference.rtc.getResourceBySSRC(ssrc);
+            if (!id) {
+                return;
+            }
+
+            // ssrc to resolution map for user id
+            var idResolutions = id2resolution[id] || {};
+            idResolutions[ssrc] = resolution;
+
+            id2resolution[id] = idResolutions;
+        });
+
+        stats.resolution = id2resolution;
+
+        conference.eventEmitter.emit(
+            JitsiConferenceEvents.CONNECTION_STATS, stats);
+    });
+
+    conference.on(JitsiConferenceEvents.TRACK_MUTE_CHANGED,
+        function (track) {
+            if(!track.isLocal())
+                return;
+            var type = (track.getType() === "audio")? "audio" : "video";
```

Comment:
I'll remove this check because anyway track.getType() should return MediaType.AUDIO or MediaType.Video.